django/forms/boundfield.py

```diff
@@ -278,6 +278,13 @@ def build_widget_attrs(self, attrs, widget=None):
                 attrs["required"] = True
         if self.field.disabled:
             attrs["disabled"] = True
+        id_for_label = self.id_for_label
+        if self.field.help_text and id_for_label:
+            helptext_id = '%s_helptext' % id_for_label
+            if 'aria-describedby' in widget.attrs:
+                attrs['aria-describedby'] = f"{helptext_id} {attrs['aria-describedby']}"
```

Comment:
Use `widget.attrs` instead of `attrs`. Also, f-strings should use only plain variable and property access (this guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style)):
```suggestion
            if described_by := widget.attrs.get("aria-describedby")
                attrs["aria-describedby"] = f"{helptext_id} {described_by}"
```

Reply:
I implemented this change and unit tests are passing now. However, 21 checks are still failing. See [this failure](https://djangoci.com/job/pr-mariadb/database=mysql,label=mariadb,python=python3.11/17866/testReport/junit/forms_tests.tests.test_forms/FormsTestCase/test_custom_describedby/) for example where the error is 
`  File "/home/jenkins/workspace/pr-mariadb/database/mysql/label/mariadb/python/python3.11/django/forms/boundfield.py", line 285, in build_widget_attrs
    attrs['aria-describedby'] = f"{helptext_id} {attrs['aria-describedby']}"
                                                 ~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'aria-describedby'.` 
I am not sure why this is happening because I implemented your suggested change in this [commit](https://github.com/django/django/pull/16185/commits/6b2c1ee20fa052e9ec9aeaa7a451a2d9049ffbfb)

Reply:
> Update: I used `widget.attrs` instead of `attrs` and I'm not getting KeyError when running unit tests now. However, 21 checks are still failing. See [this failure](https://djangoci.com/job/pr-mariadb/database=mysql,label=mariadb,python=python3.11/17866/testReport/junit/forms_tests.tests.test_forms/FormsTestCase/test_custom_describedby/) for example where the error is ` File "/home/jenkins/workspace/pr-mariadb/database/mysql/label/mariadb/python/python3.11/django/forms/boundfield.py", line 285, in build_widget_attrs attrs['aria-describedby'] = f"{helptext_id} {attrs['aria-describedby']}" ~~~~~^^^^^^^^^^^^^^^^^^^^ KeyError: 'aria-describedby'.` I am not sure why this is happening because I implemented your suggested change in this [commit](https://github.com/django/django/pull/16185/commits/6b2c1ee20fa052e9ec9aeaa7a451a2d9049ffbfb) @felixxm

It looks like an issue with CI fetching the current state of your branch :thinking: I rebased and force-pushed to your branch, it should help. Use `git pull --rebase` locally to get the current state.